src/SignalR/server/Core/src/HubConnectionHandler.cs

```diff
@@ -96,7 +96,7 @@ public override async Task OnConnectedAsync(ConnectionContext connection)
             var connectionContext = new HubConnectionContext(connection, contextOptions, _loggerFactory);
 
             var resolvedSupportedProtocols = (supportedProtocols as IReadOnlyList<string>) ?? supportedProtocols.ToList();
-            if (!await connectionContext.HandshakeAsync(handshakeTimeout, resolvedSupportedProtocols, _protocolResolver, _userIdProvider, _enableDetailedErrors))
+            if (!await connectionContext.HandshakeAsync(handshakeTimeout, resolvedSupportedProtocols, _maximumMessageSize, _protocolResolver, _userIdProvider, _enableDetailedErrors))
```

Comment:
We could put `MaximumMessageSize` in `HubConnectionContextOptions` instead if that's prefered.

Reply:
Yes please.